### PR TITLE
Upgrade to ECL 2.0 syntax

### PR DESCRIPTION
**Pull request by cmark**
This PR upgrades the syntax to the latest ECL 2.0 draft specification (as of Feb 15, 2022).

### CODE REVIEW

File: com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext [hunk 1]
```diff
@@ -297,6 +309,16 @@ CaseSignificanceFilter:
 FilterValue:
 	FilteredExpressionConstraint | EclConceptReferenceSet;
 
+// History Supplement from ECL 2.0
+Supplement:
+	HistorySupplement;
+
+HistorySupplement:
+	DOUBLE_CURLY_OPEN PLUS HISTORY_KEYWORD {HistorySupplement} history=(HistoryProfile | NestedExpression)? DOUBLE_CURLY_CLOSE;
```
Comment (apeteri) [inline com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext:317]: Have we turned off `WS` hiding by default? I don't know whether the grammar allows eg. `HISTORY - MIN` or not. 🤔

Comment (cmark) [reply com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext:317]: See the top line of the grammar definition: `hidden(WS, ML_COMMENT, SL_COMMENT)`
You can put whitespace and comment basically everywhere.

File: com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/ecl/HistoryProfileType.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Arrays;
+
+/**
+ * @since 2.0
+ */
+public enum HistoryProfileType {
+
+	MIN("match"), 
+	MOD("wild"),
+	MAX("regex");
```
Comment (apeteri) [inline com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/ecl/HistoryProfileType.java:27]: These literals are suspicious. 😃

Comment (cmark) [reply com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/ecl/HistoryProfileType.java:27]: Oh god 🤦‍♂️

**Review by apeteri — APPROVED** [review]
Please fix the constructors of `HistoryProfileType`, then the PR is good to go! 👍

File: com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext (hunk 1 above)
Comment (apeteri) [reply com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext:317]: We temporarily un-hide `WS` for unquoted search terms, I think.

Comment (cmark) [reply com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/Ecl.xtext:317]: Fixed

File: com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/ecl/HistoryProfileType.java (hunk 2 above)
Comment (cmark) [reply com.b2international.snomed.ecl/src/com/b2international/snomed/ecl/ecl/HistoryProfileType.java]: Fixed

**Pull request merged**